### PR TITLE
Add Github Actions workflow for deploying frontend

### PR DESCRIPTION
**Pull request by chaitanyabaranwal**
## Summary

This PR makes a few changes:

* **Decouples the workflows for pull request and push**: Since we need a deploy only for changes pushed to master, I needed to create different workflows for a PR and a push against master. The GCS upload step is only present in the push workflow.
* **Creates a new Google IAM service account**: I followed the [steps here](https://github.com/google-github-actions/auth#setup) to create a new service account for uploading files to GCS.

## Testing
 
All commits except the last commit were tested with a "test" workflow. The latest run of the test workflow passed: https://github.com/bettersg/call-home/actions/runs/4636034633.

### CODE REVIEW

**Comment by glencbz** [comment]
> Creates a new Google IAM service account 

@chaitanyabaranwal could you add the service account + permissions to Terraform? If you've already created the SA in the UI, you can `terraform import` it.

**Comment by glencbz** [comment]
> https://github.com/google-github-actions/auth#setup

Oh nice! This uses OIDC instead of embedding secrets and stuff. This looks great for security and maintainability. Unfortunately this looks like a **lot** of setup
( it has a workload identity pool and everything, I can barely remember what that is).

I'd ordinarily insist on having everything set up in Terraform, but since it's already useful and it'll take a while for us to fully understand what we need, I'm happy to take this without the Terraform stuff and figure that out later.

File: .github/workflows/pull_request.yml [hunk 1]
```diff
@@ -36,9 +31,9 @@ jobs:
...
       working-directory: apps/super-app/
-    - name: Build
-      run: npm run build
-      working-directory: apps/super-app/
     - name: Test
       run: npm test
       working-directory: apps/super-app/
+    - name: Build
+      run: npm run build
+      working-directory: apps/super-app/
```
Comment (glencbz) [inline .github/workflows/pull_request.yml:39]: Why the reordering?

File: .github/workflows/push.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    - name: Test
+      run: npm test
+      working-directory: apps/super-app/
+    - name: Build
+      run: npm run build
+      working-directory: apps/super-app/
+    - name: 'Authenticate'
+      uses: 'google-github-actions/auth@v1'
+      with:
+        workload_identity_provider: 'projects/427853968081/locations/global/workloadIdentityPools/github-actions-pool/providers/github-actions-provider'
```
Comment (glencbz) [inline .github/workflows/push.yml:46]: Could we add a TODO to add this to the terraform setup?

File: .github/workflows/push.yml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        service_account: 'github-actions-service-account@call-home-staging.iam.gserviceaccount.com'
```
Comment (glencbz) [inline .github/workflows/push.yml:47]: Could we add a TODO to add this to the terraform setup?

File: .github/workflows/pull_request.yml (hunk 1 above)
Comment (chaitanyabaranwal) [reply .github/workflows/pull_request.yml:39]: I thought it makes more sense to build once testing is finished!

File: .github/workflows/push.yml (hunk 2 above)
Comment (chaitanyabaranwal) [reply .github/workflows/push.yml:47]: Done in [eae7305](https://github.com/bettersg/call-home/pull/167/commits/eae7305fa575ec5167b5c6c2027334737095741d)

File: .github/workflows/push.yml (hunk 3 above)
Comment (chaitanyabaranwal) [reply .github/workflows/push.yml:48]: Done in [eae7305](https://github.com/bettersg/call-home/pull/167/commits/eae7305fa575ec5167b5c6c2027334737095741d)

**Pull request merged**